### PR TITLE
Support Database for managing collection and index

### PR DESCRIPTION
**Pull request by chyezh**
Related issue #448

### CODE REVIEW

File: client/interceptor.go [hunk 1]
```diff
@@ -9,24 +9,45 @@ import (
...
 	return func(ctx context.Context, desc *grpc.StreamDesc, cc *grpc.ClientConn, method string, streamer grpc.Streamer, opts ...grpc.CallOption) (grpc.ClientStream, error) {
-		ctx = AuthenticationInterceptor(ctx, username, password)
+		ctx = authenticationInterceptor(ctx, username, password)
+		return streamer(ctx, desc, cc, method, opts...)
+	}
+}
+
+// databaseNameInterceptor appends the dbName into metadata.
+func databaseNameInterceptor(ctx context.Context, dbName string) context.Context {
+	return metadata.AppendToOutgoingContext(ctx, "database", dbName)
```
Comment (jaime0815) [inline client/interceptor.go:36]: use dbname as a metadata key instead of database

File: mocks/MilvusServiceServer.go
Comment (chyezh) [inline mocks/MilvusServiceServer.go:1]: Auto generated.

File: Makefile [hunk 2]
```diff
@@ -28,7 +28,7 @@ endif
 install-tool:
 	@mkdir -p $(INSTALL_PATH)
 	@$(INSTALL_PATH)/golangci-lint --version 2>&1 1>/dev/null || (echo "Installing golangci-lint into ./bin/" && curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(INSTALL_PATH) v1.46.2)
-	@$(INSTALL_PATH)/mockery --version 2>&1 1>/dev/null || (echo "Installing mockery v2.16.0 to ./bin/" && GOBIN=$(INSTALL_PATH)/ go install github.com/vektra/mockery/v2@v2.16.0)
+	@$(INSTALL_PATH)/mockery --version 2>&1 1>/dev/null || (echo "Installing mockery v2.26.1 to ./bin/" && GOBIN=$(INSTALL_PATH)/ go install github.com/vektra/mockery/v2@v2.26.1)
```
Comment (jaime0815) [inline Makefile:31]: Suggest keeping same version 2.16 for mockery with milvus

File: client/database.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Database represents a database in a remote Milvus cluster.
+type Database struct {
+	Name string // Database name.
+}
```
Comment (congqixia) [inline client/database.go:22]: It is suggested to put response entity in entity pkg

File: client/client_grpc_collection.go [hunk 4]
```diff
@@ -46,6 +48,14 @@ func (c *GrpcClient) connect(ctx context.Context, addr string, opts ...grpc.Dial
 	return nil
 }
 
+// UsingDatabase creates a new client with given database name.
+// The new client is safe to use and does not interfere with the original client.
+func (c *GrpcClient) UsingDatabase(ctx context.Context, dbName string) (Client, error) {
```
Comment (congqixia) [inline client/client_grpc_collection.go:53]: `Using database` changes current session database in convention.

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**